### PR TITLE
Update forger plugin to use lisk-framework testing - Closes #6130

### LINKED ISSUE #6130: Update forger plugin to use lisk-framework
### Description

Update forger plugin tests to use created testing utilities.

### Motivation

- All testing utilities should be used in the framework to test the usability of the functions

### Acceptance Criteria

- Plugin functional tests are using the utility functions created

### PR DESCRIPTION
**Pull request by ishantiw**
### What was the problem?

This PR resolves #6130 

### How was it solved?

- Update application environment to be a class and with below usage

```ts
appEnv = new ApplicationEnv(config);
appEnv.startApplication() // starts both app and IPC client
appEnv.closeApplication() // disconnects ipc client and shutdown application
// usage
appEnv.application
appEnv.client
// OR
appEnv.getLastblock
appEnv.networkIdentifier
```

- Update forgerplugin to use lisk-framework testing

### How was it tested?

`npm run test:functional`

### CODE REVIEW

File: framework-plugins/lisk-framework-forger-plugin/test/utils/application.ts [hunk 1]
```diff
@@ -87,33 +77,24 @@ export const createApplication = async (
...
-		await app['_blockchainDB'].clear();
-		const forgerPluginInstance = getForgerPlugin(app);
-		await forgerPluginInstance['_forgerPluginDB'].clear();
-	}
-
-	await app.shutdown();
+	await appEnv.stopApplication(options);
 };
 
 export const getURL = (url: string, port = forgerApiPort): string =>
```
Comment (Incede) [inline framework-plugins/lisk-framework-forger-plugin/test/utils/application.ts:100]: I think we can remove this function here since it's not being used

**Review by shuse2 — APPROVED** [review]

**Review by Incede — APPROVED** [review]

File: framework-plugins/lisk-framework-forger-plugin/test/utils/application.ts [hunk 2]
```diff
@@ -87,51 +77,27 @@ export const createApplication = async (
...
-		// Remove pre-existing data
-		fs.removeSync(path.join(rootPath, label).replace('~', os.homedir()));
-	}
+	const appEnv = new testing.ApplicationEnv({
+		modules: [TokenModule, SequenceModule, KeysModule],
+		config,
+		plugins: [ForgerPlugin],
+		genesisBlock,
+	});
+	// const appEnv = await testing.getApplicationEnv();
```
Comment (mitsuaki-u) [inline framework-plugins/lisk-framework-forger-plugin/test/utils/application.ts:86]: can remove this comment

File: framework/src/testing/app_env.ts [hunk 3]
```diff
@@ -15,60 +15,112 @@
...
+		if (this._application.config.rpc.enable && this._application.config.rpc.mode === 'ipc') {
+			await this._ipcClient.disconnect();
+		}
+		await this._application.shutdown();
+	}
 
-	// eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-	const dataPath = pathResolve(`${homedir()}/.lisk/${label}`);
-	const apiClient = createIPCClient(dataPath);
+	public async waitNBlocks(n = 1): Promise<void> {
```
Comment (mitsuaki-u) [inline framework/src/testing/app_env.ts:90]: this function already exists in testing framework utilities

**Review by mitsuaki-u — CHANGES_REQUESTED** [review]
LGTM some minor comments

**Review by mitsuaki-u — APPROVED** [review]

**Pull request merged**